zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -47,17 +51,20 @@ where
         .or_else(|| params.activation_height(NetworkUpgrade::Sapling))
         .ok_or(Error::SaplingNotActive)?;
 
-    let outputs = decrypt_transaction(params, height, tx, &extfvks);
-    if outputs.is_empty() {
-        Ok(())
-    } else {
-        data.store_received_tx(&ReceivedTransaction {
-            tx,
-            outputs: &outputs,
-        })?;
+    let sapling_outputs = decrypt_transaction(params, height, tx, &extfvks);
 
-        Ok(())
+    if !(sapling_outputs.is_empty() && tx.transparent_bundle().iter().all(|b| b.vout.is_empty())) {
```

Comment:
Hm, this strikes me as a bug; it relies upon the caller to only pass transactions that involve the wallet in some way. The current behavior isn't necessarily harmful, because we expect the caller to have already detected that the transaction is related to the wallet (otherwise they wouldn't have downloaded the full transaction data) but I've added a ticket to correct this: https://github.com/zcash/librustzcash/issues/497